arch/riscv/Kconfig

```diff
@@ -206,6 +206,18 @@ config RISCV_GENERIC_TOOLCHAIN
 	  Allow SOCs that have custom extended riscv ISA to still
 	  compile with generic riscv32 toolchain.
 
+config RISCV_KERNEL_IN_USER_MODE
```

Comment:
Since we already have, CONFIG_USERSPACE, which can be used for RISC-V, why not adding just the config below, for the S mode ? So Zephyr "boots" in S mode and eventually, if needed, USERSPACE can be turned on.

Reply:
regarding Supversior mode, this would be possible.

regarding User mode, its still possible to run zephyr in User mode and still have CONFIG_USERSPACE active, then the user threads would still use the syscall to enter the kernel (which is also running in RISCV User Mode).

Reply:
@achech,

the CONFIG_USERSPACE is orthogonal to this, IMO. It is about Zephyr OS itself - if it have user mode threads or kernel mode only. So technically we can have CONFIG_USERSPACE=y (so Zephyr would have kernel and user threads) but run it in RV machine and supervisor mode.